src/EditorFeatures/Core/Preview/AbstractPreviewFactoryService.cs

```diff
@@ -670,21 +672,15 @@ private async ValueTask<ITextBuffer> CreateNewPlainTextBufferAsync(TextDocument
                 // won't happen.
                 originalBuffer.DeleteSpans(0, originalBuffer.CurrentSnapshot.SpanCount);
                 changedBuffer.DeleteSpans(0, changedBuffer.CurrentSnapshot.SpanCount);
-
-                leftWorkspace?.Dispose();
-                leftWorkspace = null;
-
-                rightWorkspace?.Dispose();
-                rightWorkspace = null;
             };
 
             if (_editorOptionsService.GlobalOptions.GetOption(SolutionCrawlerRegistrationService.EnableSolutionCrawler))
             {
-                leftWorkspace?.EnableSolutionCrawler();
-                rightWorkspace?.EnableSolutionCrawler();
+                leftWorkspace?.Target.EnableSolutionCrawler();
+                rightWorkspace?.Target.EnableSolutionCrawler();
             }
 
-            return new DifferenceViewerPreview(diffViewer);
+            return differenceViewer.TryAddReference()!;
```

Comment:
for my own edification, this should always succeed, yes? (i presume that's also why we have the `!`).  I feel ike it would be nice on these RefCountedDisposables to have a method (or extension) that adds and asserts/throws if that somehow failed (Similar to GetRequiredDocument vs GetDocument).

Reply:
⏱️ Will add a `AddReference()` method

Reply:
➡️ Implemented in b228912c0d883c2f39f5413ace7f46cbd9c279b4